apis/metal3.io/v1alpha1/doc.go

```diff
@@ -0,0 +1,19 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1alpha1 contains API Schema definitions for the metal3.io v1alpha1 API group
+// +kubebuilder:object:generate=true
+// +groupName=metal3.io
```

Comment:
These comments already exist [here](https://github.com/metal3-io/baremetal-operator/blob/main/apis/metal3.io/v1alpha1/groupversion_info.go#L16-L18). Do we need 2 or should we delete those ones?

Reply:
I left them there because I'm unsure if someone might be relying on those.
IMO they should be just in `doc.go`, but it might be better to defer the removal or leave them there after all.

I tried to find an example of a similar situation in a different project from the metal3-io org.
You can find something similar going on in [the cluster-api-provider project](https://github.com/metal3-io/cluster-api-provider-metal3/tree/v1.3.5/api/v1alpha5).
The `v1alpha5` types contain both [`doc.go`](https://github.com/metal3-io/cluster-api-provider-metal3/blob/b75132f98da6b1954c8e2ca90ec19852d9b916b5/api/v1alpha5/doc.go#L23) and [`groupversion_info.go`](https://github.com/metal3-io/cluster-api-provider-metal3/blob/b75132f98da6b1954c8e2ca90ec19852d9b916b5/api/v1alpha5/groupversion_info.go#L22) files with the duplicate comment tags.
The same happens for the more recent `v1beta1`: 
- https://github.com/metal3-io/cluster-api-provider-metal3/blob/5dceef85d4794a647161e2bfac57b66554e573b2/api/v1beta1/doc.go#L22
- https://github.com/metal3-io/cluster-api-provider-metal3/blob/5dceef85d4794a647161e2bfac57b66554e573b2/api/v1beta1/groupversion_info.go#L22

> TBH I have no idea what any of this does.

The comment tags are mainly used for code and spec generation.
Hence the breaking changes in the generated code, which is now actually considering the tag that was previously ignored since it was only in the `groupversion_info.go` file.

In my case I'm using them to generate an OpenAPI spec with [kube-openapi](https://github.com/kubernetes/kube-openapi) and with that spec generate reliable Java types :exploding_head: (https://github.com/fabric8io/kubernetes-client/pull/6417).

Reply:
Looks fine to keep both.